src/node.js

```diff
@@ -22,6 +22,50 @@
 
     EventEmitter.call(process);
 
+    const traceWarnings = process.traceProcessWarnings;
+    const noDeprecation = process.noDeprecation;
+    const traceDeprecation = process.traceDeprecation;
+    const throwDeprecation = process.throwDeprecation;
+    const prefix = `(${process.release.name}:${process.pid}) `;
+    if (!process.noProcessWarnings) {
+      process.on('warning', (warning) => {
+        if (warning instanceof Error) {
+          const isDeprecation = warning.name === 'DeprecationWarning';
+          if (isDeprecation && noDeprecation) return;
+          const trace = traceWarnings || (isDeprecation && traceDeprecation);
+          if (trace && warning.stack) {
+            console.error(`${prefix}${warning.stack}`);
+          } else {
+            var toString = warning.toString;
+            if (typeof toString !== 'function')
+              toString = Error.prototype.toString;
```

Comment:
Yeah, can't do that since it could be different with every `Error` object passed in.